src/main/scala/org/broadinstitute/dsde/firecloud/service/LibraryService.scala

```diff
@@ -83,7 +83,15 @@ class LibraryService (protected val argUserInfo: UserInfo, val rawlsDAO: RawlsDA
               // between the time we retrieved them and here, where we update them.
               val allOperations = generateAttributeOperations(workspaceResponse.workspace.attributes, userAttrs,
                 k => k.namespace == AttributeName.libraryNamespace && k.name != LibraryService.publishedFlag.name)
-              rawlsDAO.patchWorkspaceAttributes(ns, name, allOperations) map (RequestComplete(_))
+              val republish = workspaceResponse.workspace.attributes.contains(publishedFlag) &&
+                workspaceResponse.workspace.attributes.get(publishedFlag).fold(false)(_.asInstanceOf[AttributeBoolean].value)
```

Comment:
we should be the only ones creating the publish flag.... and when someone sets metadata via swagger, it validates against the attribute file... i think we're ok